### PR TITLE
Split integration test suite into non-invasive integration and invasive functional suite

### PR DESCRIPTION
**Pull request by FelixJacobi**
This allows us to run the `CurlTransportTest` in Workflows to cover the class correctly.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -94,9 +94,15 @@
       "BigBlueButton\\": ["src", "tests/integration"]
     }
   },
+  "autoload-dev": {
+    "psr-4": {
+      "BigBlueButton\\Tests\\Functional\\": ["tests/functional"]
```
Comment (FelixJacobi) [inline composer.json:99]: Note: My plan is to do such Namespace restructuring also for `unit` and `integration` tests later. The reason is to improve DX on IDE integration (e.g. for PHPStorm, which automatically derives the namespace from the composer.json)

File: composer.json [hunk 2]
```diff
@@ -111,7 +117,7 @@
         "vendor/bin/php-cs-fixer fix --dry-run --allow-risky=yes"
       ],
       "pre-push": [
-        "vendor/bin/phpunit --testsuite=\"BigBlueButton test suite\"",
+        "vendor/bin/phpunit --testsuite=\"BigBlueButton unit test suite,BigBlueButton integration test suite\"",
```
Comment (FelixJacobi) [inline composer.json:120]: Hm, does not work yet.

File: phpunit-integration.xml.dist [hunk 3]
```diff
@@ -1,26 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
```
Comment (FelixJacobi) [inline phpunit-integration.xml.dist:1]: Do we need this file anymore? The test suites are all defined in `phpunit.xml.dist`.

File: tests/integration/Http/Transport/CurlTransportTest.php [hunk 4]
```diff
@@ -121,6 +121,10 @@ public function testRequestWithoutPayload(): void
 
     public function testRequestWithCookie(): void
     {
+        if (\PHP_VERSION_ID >= 80000) {
+            self::markTestSkipped('Broken on PHP 8. To be fixed in https://github.com/littleredbutton/bigbluebutton-api-php/pull/78.');
```
Comment (FelixJacobi) [inline tests/integration/Http/Transport/CurlTransportTest.php:125]: To be reverted in #78.

File: composer.json (hunk 2 above)
Comment (FelixJacobi) [reply composer.json:120]: Never mind. Forget to run `./vendor/bin/cghooks update`.

File: phpunit-integration.xml.dist (hunk 3 above)
Comment (sualko) [reply phpunit-integration.xml.dist:1]: I think we can remove it. It's not referenced anywhere.

**Review by sualko — APPROVED** [review]

**Pull request merged**